src/Features/Core/Portable/CodeRefactorings/AddMissingImports/AbstractAddMissingImportsRefactoringProvider.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.PasteTracking;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.AddMissingImports
+{
+    internal abstract class AbstractAddMissingImportsRefactoringProvider : CodeRefactoringProvider
+    {
+        private readonly IPasteTrackingService _pasteTrackingService;
+        protected abstract string CodeActionTitle { get; }
+
+        public AbstractAddMissingImportsRefactoringProvider(IPasteTrackingService pasteTrackingService)
+        {
+            _pasteTrackingService = pasteTrackingService;
+        }
+
+        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var document = context.Document;
+
+            // Currently this refactoring requires the document to have a pasted text span.
+            if (!_pasteTrackingService.TryGetPastedTextSpan(document, out var textSpan))
```

Comment:
this is a refactoring.  so async can be used.  so instead of passing the document, asynchornously acquire and pass the SourceTextContainer.  Then, in the tracking service, there will be no need for TryGetText.